src/test/java/software/amazon/encryption/s3/S3EncryptionClientRangedGetCompatibilityTest.java

```diff
@@ -37,16 +41,47 @@
 public class S3EncryptionClientRangedGetCompatibilityTest {
 
     private static SecretKey AES_KEY;
+    private static KeyPair RSA_KEY_PAIR;
 
     @BeforeAll
     public static void setUp() throws NoSuchAlgorithmException {
         KeyGenerator keyGen = KeyGenerator.getInstance("AES");
         keyGen.init(256);
         AES_KEY = keyGen.generateKey();
+
+        KeyPairGenerator keyPairGen = KeyPairGenerator.getInstance("RSA");
+        keyPairGen.initialize(2048);
+        RSA_KEY_PAIR = keyPairGen.generateKeyPair();
     }
 
-    @Test
-    public void AsyncAesGcmV3toV3RangedGet() {
+    static Object[] keyMaterialProvider() {
+        return new Object[] {
+                AES_KEY,
+                RSA_KEY_PAIR,
+        };
+    }
+
+    private static S3EncryptionClient.Builder addKeyMaterialToClient(S3EncryptionClient.Builder builder, Object keyMaterial) {
+        if (keyMaterial instanceof  SecretKey) {
+            builder.aesKey((SecretKey) keyMaterial);
+        } else if (keyMaterial instanceof KeyPair) {
+            builder.rsaKeyPair((KeyPair) keyMaterial);
+        }
+        return builder;
+    }
+
+    private static S3AsyncEncryptionClient.Builder addKeyMaterialToClient(S3AsyncEncryptionClient.Builder builder, Object keyMaterial) {
+        if (keyMaterial instanceof  SecretKey) {
+            builder.aesKey((SecretKey) keyMaterial);
+        } else if (keyMaterial instanceof KeyPair) {
+            builder.rsaKeyPair((KeyPair) keyMaterial);
+        }
+        return builder;
+    }
+
+    @ParameterizedTest
```

Comment:
I can't put comments on lines which have no diff, but we should parameterize the CBC based tests as well, for extra coverage. Particularly given my other comment above.